assets/src/edit-story/app/story/useStoryReducer/reducers/test/utils.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Internal dependencies
+ */
+import {
+	intersect,
+	isInsideRange,
+	moveArrayElement,
+	getAbsolutePosition,
+	objectWithout,
+} from '../utils';
+import { LAYER_DIRECTIONS } from '../../../../../constants';
+
+const ABC = [ 'A', 'B', 'C' ];
+const BCD = [ 'B', 'C', 'D' ];
+const ABCD = [ 'A', 'B', 'C', 'D' ];
+const D = [ 'D' ];
+
+describe( 'intersect', () => {
+	it( 'should return first element if only one given', () => {
+		const result = intersect( ABC );
+
+		expect( result ).toStrictEqual( ABC );
+	} );
+
+	it( 'should return intersection set if multiple elements given', () => {
+		const shouldBeBC = intersect( ABC, BCD );
+		expect( shouldBeBC ).toStrictEqual( [ 'B', 'C' ] );
+
+		const shouldBeD = intersect( ABCD, BCD, D );
+		expect( shouldBeD ).toStrictEqual( [ 'D' ] );
+
+		const shouldBeEmpty = intersect( ABC, BCD, D );
+		expect( shouldBeEmpty ).toStrictEqual( [] );
+	} );
+} );
+
+describe( 'isInsideRange', () => {
+	it( 'should function as expected', () => {
+		const isSingleDigit = ( number ) => isInsideRange( number, 0, 9 );
+
+		expect( isSingleDigit( 0 ) ).toStrictEqual( true );
+		expect( isSingleDigit( 3.2 ) ).toStrictEqual( true );
+		expect( isSingleDigit( 9 ) ).toStrictEqual( true );
+
+		expect( isSingleDigit( -0.1 ) ).toStrictEqual( false );
+		expect( isSingleDigit( 10 ) ).toStrictEqual( false );
+
+		expect( isSingleDigit( Number.POSITIVE_INFINITY ) ).toStrictEqual( false );
+		expect( isSingleDigit( Number.NEGATIVE_INFINITY ) ).toStrictEqual( false );
+		expect( isSingleDigit( Number.NaN ) ).toStrictEqual( false );
+	} );
+} );
+
+describe( 'moveArrayElement', () => {
+	it( 'should move element forwards', () => {
```

Comment:
Could we add a test for trying moving the element outside of limits?

Reply:
Good idea - done in d6c410c9.